### ISSUE TITLE
Ability to add segments to a track

### ISSUE THREAD

**Issue by GrazianoCapelli** [BasicAirData/GPSLogger#68]
Today the tracks are made of a single segment.
If you pause / resume recording, the path will continue to be recorded into the same segment.
A pause is treated like a momentary "loss of signal" of the GPS.

We could add the ability to record multiple segments.

Basically, we could add for example a button to the upper toolbar (visible maybe only in pause state), to add a new segment to the current track. Starting from the click of this button, the new Trackpoints/Placemarks will be recorded into the new segment.

An indication of the number of segment should be present on the UI.

We start here to collect ideas, mockups and suggestions.
Comments are welcome!

**Comment by ygoe** [comment]
May I ask what's the use of multiple segments per track? I see there's that structure in GPX files, but I never had any use for them. I just need a list of time-stamped locations.

Anyway, if segment splitting is an option, I could imagine that goes on the main screen next to the record/pause/stop buttons, just as the "add named placemark" button. See my https://github.com/BasicAirData/GPSLogger/issues/88#issuecomment-706522214 for screenshots/suggestions. That would then make 4 buttons: Add placemark, pause, stop, split (in whatever order).

Caution: Track splitting is as final as stopping the recording. Some applications might not be able to handle multiple segments. The same safety considerations should apply as for stopping the recording.

**Comment by phicou44** [comment]
Hello,

Please allow me to disagree with @ygoe.

I often use GPS Logger during road trips to record walks at different places located far from each other. For example, I go somewhere where I start the recording, I do a walk, pause GPS logger, move to another location at 10 km/miles, then I restart the recording, and it can occurs multiple times. The final effect when I display the track is to have very long strait lines between the "real" parts of the track, this is very annoying, see 1st capture below. With segments, the result would be what is shown in 2nd capture, better IMHO.

Some could suggest to start/stop recording different tracks rather than pausing, but in my case having only one GPX file is easier to handle and avoids some problems.

It should still be noted that if pause is set and unset at the same location (e.g. a lunch during a walk), there is no need to create a new segment for that.

So, I would suggest, when restarting recording after a pause, to create a new segment only if the the location is far from where the pause was set, for example 1/2 km/mile. That way, there is no need to add supplementary buttons to the UI. However, the creation of multiple segments or the gap threshold could be controlled by preferences, but this is not mandatory.

Technically speaking, it seems to me that this would be easy to implement :-)

Without segments:
![Capture_1](https://github.com/BasicAirData/GPSLogger/assets/50493907/140552c0-3c5e-47aa-8cec-36be0e8fb362)

With segments:
![Capture_2](https://github.com/BasicAirData/GPSLogger/assets/50493907/1d47800a-397d-4689-b466-ab0e7e20c51b)

**Comment by uniform641** [comment]
I agree with @phicou44 that it's much more convenient with track segment.

I use GPSLogger to record my trips. The tracks are often interrupted by some events:
1. GPS signal loss due to shelter above the path. e.g. taking subway, driving through tunnel, walking through building.
2. Activities needless to be recorded. e.g. having meal along the trip.

I think the record before and after the events should still be considered as one track. And now these segments are just connected by ugly straight lines, which actually do not record the real tracks.
